src/UmbracoFileSystemProviders.Azure/AzureFileSystem.cs

```diff
@@ -160,6 +161,12 @@ internal AzureFileSystem(string containerName, string rootUrl, string connection
         public bool UseDefaultRoute { get; }
 
         /// <summary>
+        /// Gets or sets func to calculate application virtual path
+        /// </summary>
+        public Func<string> GetApplicationVirtualPath { get; internal set; } = () => HttpRuntime.AppDomainAppVirtualPath;
```

Comment:
because I cannot override this in unit tests. I'm not 100% sure that HttpRuntime.AppDomainAppVirtualPath is set correctly on the constructor. I will check it